src/strategy/mod.rs

```diff
@@ -28,13 +42,24 @@ pub(crate) enum UpdateStrategy {
 impl UpdateStrategy {
     /// Try to parse config inputs into a valid strategy.
     pub(crate) fn with_config(cfg: inputs::UpdateInput, identity: &Identity) -> Fallible<Self> {
-        let strategy = match cfg.strategy.as_ref() {
+        let strategy_name = cfg.strategy.clone();
+        let strategy = match strategy_name.as_ref() {
             "fleet_lock" => UpdateStrategy::new_fleet_lock(cfg, identity)?,
             "immediate" => UpdateStrategy::new_immediate()?,
             "periodic" => UpdateStrategy::new_periodic(cfg)?,
             "" => UpdateStrategy::default(),
             x => bail!("unsupported strategy '{}'", x),
         };
+
+        // Export info-metrics with details about current strategy.
+        STRATEGY_MODE
+            .with_label_values(&[strategy_name.as_ref()])
+            .set(1);
```

Comment:
I'm guessing this might be related to how `register_int_gauge_vec` works but when seeing a `zincati_updates_strategy_mode{strategy="immediate"} 1` the `1` after the current strategy makes me wonder if there's second meaning behind the integer other than the meaning of `enabled`. Just a thought definitely not a blocker.

Reply:
Those are info metrics, basically just a boolean where the interesting information is encoded via label values (i.e. the `immediate` here). The following article describes well how they are used when querying and aggregating: https://www.robustperception.io/exposing-the-software-version-to-prometheus